lib/ConstExtract/ConstExtract.cpp

```diff
@@ -148,6 +146,16 @@ static std::shared_ptr<CompileTimeValue> extractCompileTimeValue(Expr *expr) {
       return std::make_shared<ArrayValue>(elementValues);
     }
 
+    case ExprKind::Dictionary: {
+      auto dictionaryExpr = cast<DictionaryExpr>(expr);
+      std::vector<std::shared_ptr<TupleValue>> tuples;
+      for (auto elementExpr : dictionaryExpr->getElements()) {
+        auto elementValue = extractCompileTimeValue(elementExpr);
+        tuples.push_back(std::static_pointer_cast<TupleValue>(elementValue));
```

Comment:
Thanks for making the change to store tuples. Seems more straightforward now.

My only other question: Do you think we need to check the `elementValue->getKind()` here? On one hand, we _know_ it is a tuple, but if that was ever not the case I think this may crash. I think this is fine as-is, but wanted to raise the thought.

Reply:
Good thought, I'll wrap it in a guard and only cast if it's the correct type.